src/alfasim_sdk/_internal/hook_specs.py

```diff
@@ -919,50 +919,147 @@ def update_particle_diameter_of_solids_fields(
+        {
+            const char* plugin_id = get_plugin_id()
+            errcode = alfasim_sdk_api.get_thread_id(ctx, &thread_id);
+            if (errcode != OK) {
+                return errcode;
+            }
+            // MyStruct is a developer defined struct to hold
+            // all important information for plugin hooks.
+            MyStruct* data = nullptr;
+            errcode = alfasim_sdk_api.get_plugin_data(ctx, (void**) &data, plugin_id, thread_id);
+            if (errcode != OK) {
+                return errcode;
+            }
+
+            // compute the slip velocity using your own correlation
+            for (int i = 0; i < n_faces; ++i){
+                U_slip[i] = slip_velocity[i];
+            }
+
+            return OK;
+        }
     """
 
 
-def calculate_slurry_viscosity(
-    ctx: "void*", alpha_f: "void*", mu_f: "void*", mu_f_layer: "void*"
+def calculate_relative_slurry_viscosity(
+    ctx: "void*",
+    mu_r: "double*",
+    solid_field_index: "int",
+    layer_index: "int",
+    n_faces: "int",
 ) -> "int":
     """
-    Internal simulator hook to calculate slurry viscosity of layer(s).
+    **c++ signature** : ``HOOK_RELATIVE_SLURRY_VISCOSITY(void* ctx, double* mu_r, int solid_field_index, int layer_index, int n_faces)``
 
-    :param ctx: ALFAsim's plugins context
-    :param alpha_f: Fields Volume fractions on faces,
-    :param mu_f: Field viscosities on faces,
-    :param mu_f_layer: Layer Viscosities on faces,
+    Internal `hook` to calculate the relative slurry viscosity in the Solids Model.
+    This `hook` will be used in the Solids model to calculate the effective viscosity of the slurry
+    (solid field + continuous field).
+
+    The relative slurry viscosity is defined by:
 
+    .. math::
+
+        \\begin{equation}
+            \\mu_r = \\frac{\\mu_{eff}}{\\mu_c}
+        \\end{equation}
+
+
+    .. rubric:: Where
+
+    :1: :math:`\\mu_r` is the relative emulsion viscosity
+    :2: :math:`\\mu_{eff}` is the effective viscosity
+    :3: :math:`\\mu_c` is the viscosity of the continuous field
+
+    The output variable ``mu_r`` is the relative slurry viscosity with size equal to ``n_faces`` and it
+    is dimensionless.
+    The ``solid_field_index`` is the index of the dispersed solid field.
+    The ``layer_index`` is the index of the layer and continuous field in the layer(in which the solid
+    field is dispersed).
+
+    .. Note::
+        It is important to know that the calculations for viscosity is performed over the faces (see
+        ``n_faces`` param). For that, any variable that should be retrieved using :cpp:func:`get_simulation_array`
+        must be use value ``FACE`` in the :cpp:enum:`GridScope` param.
+
+    This `hook` allows the implementation of the relative slurry viscosity correlation. Once the plugin
+    installed it is important to be selected in the solids model configuration inside the Physics data
+    tree in the ALFAsim application in order to be used.
```

Comment:
Is there any way we could avoid these duplicated definitions? There is a lot of repeated documentation between `HOOK_RELATIVE_SLURRY_VISCOSITY` and `HOOK_CALCULATE_SLIP_VELOCITY`.

Reply:
If you take a look at all hooks, you will see a lot of duplicated documentation.

I remember that we decided to keep it because this part of the documentation can be consulted isolated. I mean, we can read the documentation of just one Hook and no information would be missing.

However, I think we can review this.

Reply:
ok, thanks!